typing/types.mli

```diff
@@ -294,7 +294,7 @@ type type_declaration =
     (* definition level * expansion level *)
     type_loc: Location.t;
     type_attributes: Parsetree.attributes;
-    type_immediate: bool; (* true iff type should not be a pointer *)
+    type_repr: type_repr;
```

Comment:
I'm not familiar with the process so I don't know if you should do it in this PR, but some magic numbers will need to change as a consequence of this change.